### PR TITLE
issue #749 - implement image training flow

### LINKED ISSUE #749: Implement Image Training Flow
**Describe the solution you'd like**
Implement/modify the backend endpoints & UI needed to create an Image trainspace.

**Additional context**
The way that we implemented Image model training before the refactorization of the training flow can be found here: https://datasciencegt-dlp.com/img-models

This should have about the same folder structure as Tabular: All Image specific code should be in src/features/Train/features/Image, split the code into components, constants, redux, types folders, etc. 
There are existing layout components in src/features/Train/components that would be helpful in creating the actual UI components for Image.
Existing train constants should be modified also to allow for an Image trainspace selection. 
Make sure to modularize Image also by adding an index file to handle exporting of elements outside of Image (Tabular also has an example of this).
Please ask questions if you have any, especially for this feature (since I made some opinionated decisions on the structure of the train code on the frontend that should either be kept or changed to maintain consistency)!

**Setup Instructions (what branch to work off of)**
Run the following commands
```
git checkout nextjs
git pull origin nextjs
git checkout -b image-train
```
FYI: If you are not able to immediately run `git checkout nextjs`, make sure you commit your changes in the current branch or run `git stash` and then execute the above commands

### PR DESCRIPTION
**Pull request by AmyCGT**

### CODE REVIEW

**Comment by AmyCGT** [comment]
https://github.com/DSGT-DLP/Deep-Learning-Playground/issues/749 <- the issue

**Comment by karkir0003** [comment]
@AmyCGT put a screen recording of the flow

File: frontend/playground-frontend/src/features/Train/features/Image/components/ImageDatasetStep.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        tabs={[
+          { tabLabel: "Recently Uploaded Datasets", tabValue: "upload-dataset" },
+          { tabLabel: "Default Datasets", tabValue: "default-dataset" },
+        ]}
+        tabComponents={{
+          "upload-dataset": (
+            <UploadDatasetPanel
+              dataSource={"IMAGE"}
+              methods={uploadDatasetMethods}
+              acceptedTypes={".zip"}
```
Comment (karkir0003) [inline frontend/playground-frontend/src/features/Train/features/Image/components/ImageDatasetStep.tsx]: did u validate that only .zip files are accepted when you upload image data in image trainspace? @AmyCGT

File: frontend/playground-frontend/src/settings.tsx [hunk 2]
```diff
@@ -731,14 +731,28 @@ export const POSSIBLE_TRANSFORMS: PossibleTransform[] = [
...
+      width: {
+        index: 1,
+        parameter_name: "width",
+        min: 1,
+        max: 1000,
+        parameter_type: "number",
+      } as LayerParameter<"number">,
+      // size: {
+      //   index: 0,
+      //   parameter_name: "(H, W)",
```
Comment (karkir0003) [inline frontend/playground-frontend/src/settings.tsx]: is this needed? @AmyCGT 

i see commented out code

File: frontend/playground-frontend/src/settings.tsx [hunk 3]
```diff
@@ -897,14 +911,28 @@ export const DETECTION_TRANSFORMS: DetectionTransform[] = [
...
```
Comment (karkir0003) [inline frontend/playground-frontend/src/settings.tsx]: same nit here

**Review by karkir0003 — CHANGES_REQUESTED** [review]
address comments

@farisdurrani @dwu359 provide your inputs as well

**Comment by farisdurrani** [comment]
Hey @AmyCGT can you please fix the conflicts? I did not realize your PR had conflicts with the nextjs branch before I merged in the Remove playground-frontend PR

**Comment by farisdurrani** [comment]
The conflicts seem bad so I recommend making a copy of all the files you change, pull in the latest nextjs branch, and paste in back your changes

**Review by dwu359 — COMMENTED** [review]
image
Any way to make the inputs for Conv2d wrap around to the bottom btw?

**Review by dwu359 — COMMENTED** [review]
image
This is what shows up for me in review step btw, we're still deciding how the review step information should be displayed, but rn also display data from parameters in review step for testing purposes

**Comment by AmyCGT** [comment]
video preview

https://github.com/DSGT-DLP/Deep-Learning-Playground/assets/104514492/56b9d4a9-d180-4c3b-908f-c246b5ae38aa

**Comment by karkir0003** [comment]
@dwu359 please refer to Amy's updated PR #787 for her latest changes

**Comment by karkir0003** [comment]
closing this PR since this is old version

**Pull request closed without merging**